src/Uno.UI/UI/Xaml/UIElement.Pointers.Skia.cs

```diff
@@ -60,186 +107,402 @@ private void CoreWindow_PointerExited(CoreWindow sender, PointerEventArgs args)
-						}
-
-						if (action(parent))
-						{
-							return;
-						}
+						this.Log().Trace($"CoreWindow_PointerMoved ({args.CurrentPoint.Position}) **undispatched**");
 					}
+
+					return;
 				}
-			}
 
-			private void CoreWindow_PointerMoved(CoreWindow sender, PointerEventArgs args)
-			{
 				if (this.Log().IsEnabled(LogLevel.Trace))
 				{
-					this.Log().Trace($"CoreWindow_PointerMoved ({args.CurrentPoint.Position})");
+					this.Log().Trace($"CoreWindow_PointerMoved [{source.element.GetDebugName()}");
 				}
 
-				var pointer = new Pointer(args.CurrentPoint.PointerId, PointerDeviceType.Mouse, false, isInRange: true);
+				var routedArgs = new PointerRoutedEventArgs(args, source.element);
 
-				if (UIElement.PointerCapture.TryGet(pointer, out var capture))
+				// First raise the PointerExited events on the stale branch
+				if (source.stale.HasValue)
 				{
-					foreach (var target in capture.Targets)
+					routedArgs.CanBubbleNatively = true; // TODO: UGLY HACK TO AVOID BUBBLING: we should be able to request to bubble only up to a the root
+					var (root, stale) = source.stale.Value;
+
+					Debug.Write($"Exiting branch from (root) {root.GetDebugName()} to (leaf) {stale.GetDebugName()}\r\n");
+
+					while (stale is { })
 					{
-						var pointerArgs = new PointerRoutedEventArgs(args, target.Element);
+						routedArgs.Handled = false;
+						stale.OnNativePointerExited(routedArgs);
+						// TODO: This differs of how we behave on iOS, macOS and Android which does have "implicit capture" while pressed.
+						//		 It should only impact the "Pressed" visual states of controls.
+						stale.SetPressed(routedArgs, isPressed: false, muteEvent: true);
 
-						target.Element.OnNativePointerMove(pointerArgs);
+						if (stale == root)
+						{
+							break;
+						}
+
+						stale = stale.GetParent() as UIElement;
 					}
+					routedArgs.CanBubbleNatively = false;
 				}
-				else
+
+				// Second (try to) raise the PointerEnter on the OriginalSource
+				// Note: This won't do anything if already over.
+				routedArgs.Handled = false;
+				source.element.OnNativePointerEnter(routedArgs);
+
+				// Finally raise the event, either on the OriginalSource or on the capture owners if any
+				if (PointerCapture.TryGet(routedArgs.Pointer, out var capture))
 				{
-					PropagateEvent(args, e =>
+					foreach (var target in capture.Targets.ToArray())
 					{
-						var pointerArgs = new PointerRoutedEventArgs(args, e);
-						e.OnNativePointerMove(pointerArgs);
-					});
+						//Debug.WriteLine($"[MOVE] RE-ROUTED TO {GetTraceName(target.Element)}");
+						routedArgs.Handled = false;
+						target.Element.OnNativePointerMove(routedArgs);
+					}
+				}
+				else
+				{
+					// Note: We prefer to use the "WithOverCheck" overload as we already know that the pointer is effectively over
+					//cDebug.WriteLine($"[MOVE] {GetTraceName(source.element)}");
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6213625)